openshift-201/image-management.md

```diff
@@ -0,0 +1,434 @@
+
+
+#### Application Status
+Wait for the application to be ready and running:
+```bash
+oc get pods
+
+NAME                          READY   STATUS      RESTARTS   AGE
+hello-world-1-build           0/1     Completed   0          10m
+hello-world-85c7dc4569-njqlb  1/1     Running     0          36s
+```
+
+Use the pod name shown above (the characters after `hello-world-` will be different for you) to display the output of the logs.
+```bash
+oc logs hello-world-85c7dc4569-njqlb
+
+Hello World!  Docker Build - v1.0
+```
+
+### Update the BuildConfig
+Let's now modify the `BuildConfig` to update our image and release a new version.
+```bash
+cat <<EOF | oc apply -f -
+kind: BuildConfig
+apiVersion: build.openshift.io/v1
+metadata:
+  name: docker-build
+  labels:
+    name: docker-build
+spec:
+  output:
+    to:
+      kind: ImageStreamTag
+      name: 'hello-world:latest'
+  strategy:
+    type: Docker
+  source:
+    type: Dockerfile
+    dockerfile: |-
+      FROM registry.access.redhat.com/ubi8 
+      CMD echo 'Hello World!  Docker Build - v1.1' && exec sleep infinity
+EOF
+```
+
+#### Start the build
+```bash
+oc start-build docker-build
+```
+
+#### Follow Build
+Use the `oc logs` command to check the build logs of the `docker-build`:
+```bash
+oc logs -f bc/docker-build
+```
+<pre>
+...<em>output omitted</em>...
+Writing manifest to image destination
+Storing signatures
+...<em>output omitted</em>...
+Push successful
+</pre>
+
+#### Application Status
+Notice our `hello-world` deployment is automatically deploying our new image.  This is because our `ImageStream` tag `latest` was updated with a new image from our `BuildConfig`
+
+```bash
+oc get pods
+
+NAME                          READY   STATUS              RESTARTS   AGE
+hello-world-1-build           0/1     Completed           0          10m
+hello-world-75c89d744f-nzxpk  0/1     ContainerCreating   0          9s
+hello-world-785c7dc456-njqlb  1/1     Running             0          36s
+```
+
+Use the new pod name shown above (the characters after `hello-world-` will be different for you) to display the output of the logs once the pod has the `Running` status.
+```bash
+oc logs hello-world-75c89d744f-nzxpk
+
+Hello World!  Docker Build - v1.1
+```
+
+### ImageStream Tags
+If we look at our `ImageStream` we should see 2 items for our `latest` image tag:
+```bash
+oc get is/hello-world -o yaml | grep -A10 tags
+```
+```yaml
+  tags:
+  - items:
+    - created: "2022-05-03T18:43:28Z"
+      dockerImageReference: image-registry.openshift-image-registry.svc:5000/ad204f-dev/hello-world@sha256:43378e2447d3fd0d1a8e84ac82ae88bf269d1c60ab0de29b1dc41475d5270284
+      generation: 1
+      image: sha256:43378e2447d3fd0d1a8e84ac82ae88bf269d1c60ab0de29b1dc41475d5270284
+    - created: "2022-05-03T18:10:14Z"
+      dockerImageReference: image-registry.openshift-image-registry.svc:5000/ad204f-dev/hello-world@sha256:9f388438ee6863477829e8d95cff895654030470aba5ca55a8f76a9f291c4ce2
```

Comment:
Make a note for participants update this command with the correct namespace

Reply:
Done